### PR TITLE
Bentley et al. 2016 expression data

### LINKED ISSUE #284: Add 10.1371/journal.pcbi.1005283
Add neurotransmitter expression info to insert_worm.py and attach Evidence with the author, title, DOI (below) and one or more URLs. Any other info available for the Evidence is desired.

Machine readable data is attached to the document on the web and is available here: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1005283.s004
DOI: 10.1371/journal.pcbi.1005283

Child of #274
related to #293 (overlap in the data source)

### PR DESCRIPTION
**Pull request by clbarnes**
Resolves #284 

Data spreadsheet: https://docs.google.com/spreadsheets/d/1kCxOOKu1wAREa9VbBiWVVHh-GEC3kJk0A3YVEipPKcc/edit#gid=0

- Adds an expression_data directory in aux_data, and the expression data used in 10.1371/journal.pcbi.1005283 in a csv with the same schema as the `Modified celegans db dump.csv`
- Adds `insert_worm.upload_additional_receptors_neurotransmitters_neuropeptides_innexins()`, which recursively searches the expression_data in lexicographic sort order for `.csv` files and adds any novel data to the network
- Factors most of the insertion logic out into a separate private method, which is called by both `upload_receptors_types_neurotransmitters_neuropeptides_innexins` and `upload_additional_receptors_neurotransmitters_neuropeptides_innexins`
- Tidies up some indentation in that method (sorry about the unhelpful diff)

Still todo:

- [ ] Update data sources docs page
- [x] Add Data Integrity Test

### CODE REVIEW

**Comment by clbarnes** [comment]
I haven't been able to test this because I can't install pyopenworm through pip in either py2 or py3.

py2: I think FuXi is the problem: it requires rdflib < 3a and the the setup.py requires rdflib==4.1.2. FuXi also doesn't install its own dependencies properly (specifically ez_setup).

py3: wsgiref==0.1.2 has a py2 print statement in it.

**Comment by clbarnes** [comment]
Build failures are just due to the unit tests having the wrong expectations for the new data.

File: OpenWormData/scripts/insert_worm.py [hunk 1]
```diff
@@ -158,9 +161,11 @@ def upload_neurons():
 def get_altun_evidence():
     return parse_bibtex_into_evidence('../aux_data/bibtex_files/altun2009.bib')
 
+
```
Comment (mwatts15) [inline OpenWormData/scripts/insert_worm.py]: Changes for pep8 compliance are fine, but please don't introduce newlines unnecessarily.

File: OpenWormData/scripts/insert_worm.py [hunk 2]
```diff
@@ -197,101 +202,135 @@ def parse_bibtex_into_evidence(file_name):
...
+                    # assert the evidence on the relationship
+                    e.asserts(r)
+                    e2.asserts(r)
+
+            i += 1
 
     for neur in neurons:
-        n = NETWORK.neuron(neur)
-    print ("uploaded " + str(i) + " statements about types, receptors, innexins, neurotransmitters and neuropeptides")
+        _ = NETWORK.neuron(neur)
```
Comment (mwatts15) [inline OpenWormData/scripts/insert_worm.py]: `n` really is unused. It's okay to just remove the assignment.

Comment (clbarnes) [inline OpenWormData/scripts/insert_worm.py]: Done.

File: OpenWormData/scripts/insert_worm.py (hunk 1 above)
Comment (mwatts15) [inline OpenWormData/scripts/insert_worm.py]: To be consistent with other spacing between other methods is fine.

**Comment by clbarnes** [comment]
Build should be green, now, forgot a dash in a receptor name...

Paradoxically, I can't do the URLs for the docs page until the PR is merged but I've got the changes except the URLs in a local branch (accidentally pushed it but then --force undid it, which is why this history might look a bit squiffy) so they can go up as soon as this PR is merged and OpenWorm has taken ownership of the spreadsheet https://docs.google.com/spreadsheets/d/1kCxOOKu1wAREa9VbBiWVVHh-GEC3kJk0A3YVEipPKcc/edit#gid=0

**Comment by mwatts15** [comment]
lgtm

**Pull request merged**